lib/shrine/uploaded_file.rb

```diff
@@ -58,6 +58,10 @@ def original_filename
       # from #original_filename.
       def extension
         result = File.extname(id)[1..-1] || File.extname(original_filename.to_s)[1..-1]
+        # for shrine-url use, id may be url with query params, strip em after ?
+        if result && result.index('?') && id =~ URI::regexp
+          result = result[0..result.index('?') - 1]
+        end
```

Comment:
I think we don't necessarily have to match onto `URI.regexp`, it should be enough to just match onto `?`. I'd also prefer to make it explicit that we're patching `File.extname(id)` and not `File.extname(original_filename)`.

Reply:
I think doing something like `File.extname(URI(id).path)` might be a good solution. That should work for both non-URL and URL ids, as in the former case it will just be `URI::Generic`.

Reply:
I added the URI.regexp out of an abundance of caution -- what if it's not shrine-uri, and the original filename has a weird extension like `something.x?x`?  But I can remove it if you think it is excessive. 

I am not following you on `prefer to make it explicit that we're patching File.extname(id)`. I didn't change any code except what was necessary to keep the query params out, are you suggesting changing more code?

Reply:
I was avoiding the equivalent of `URI.parse`, because it can raise on something that is not parseable as a URI.  I am not certain all possible IDs are parseable as a URI. It would require a rescue at least, and rescues can be "expensive", as well as more complex code.

Reply:
yeah, good idea, I should add that test. and the require is a good idea too, good catch.